### PR TITLE
chore: Add tag group component

### LINKED ISSUE #29134: Create WDS tagGroup component
https://react-spectrum.adobe.com/react-spectrum/TagGroup.html

### PR DESCRIPTION
**Pull request by jsartisan**
Fixes #29134 

## Summary by CodeRabbit

- **New Features**
  - Introduced `TagGroup` and `Tag` components with features like label, description, error message, and tag removal options.
  - Added new stories for `TagGroup` to demonstrate various configurations and use cases.

- **Enhancements**
  - Updated `Switch` component with improved validation logic.
  - Enhanced `TextArea` initialization to handle undefined default values more gracefully.
  - Added `validationState` property to `HelpText` and `Field` components to support form validation states.

- **Style Updates**
  - Implemented new CSS styles for `TagGroup` and `Tag` components to improve UI consistency and interactivity.

- **Documentation**
  - Expanded Storybook documentation with examples and usage guidelines for `TagGroup` and `Tag` components.

### CODE REVIEW

File: app/client/packages/design-system/widgets/src/components/TagGroup/src/Tag.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import React from "react";
+import {
+  Tag as HeadlessTag,
+  Button as HeadlessButton,
+} from "react-aria-components";
+import type { TagProps as HeadlessTagProps } from "react-aria-components";
```
Comment (KelvinOm) [inline app/client/packages/design-system/widgets/src/components/TagGroup/src/Tag.tsx:6]: Where are the relevant changes in package.json?

File: app/client/packages/design-system/widgets/src/components/TagGroup/src/Tag.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import { getTypographyClassName } from "@design-system/theming";
+
+function Tag({ children, ...props }: HeadlessTagProps) {
+  const textValue = typeof children === "string" ? children : undefined;
+
+  return (
+    <HeadlessTag textValue={textValue} {...props} className={styles["tag"]}>
+      {({ allowsRemoving }) => (
+        <>
+          <span className={getTypographyClassName("footnote")}>{children}</span>
```
Comment (KelvinOm) [inline app/client/packages/design-system/widgets/src/components/TagGroup/src/Tag.tsx:19]: Can we pass className to the `tag` directly? Can we avoid adding span?

File: app/client/packages/design-system/widgets/src/components/TagGroup/src/TagGroup.tsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+      >
+        {children}
+      </HeadlessTagList>
+      {Boolean(description) && (
+        <HeadlessText elementType="div" slot="description">
+          <Text variant="footnote">{description}</Text>
+        </HeadlessText>
+      )}
+      {Boolean(errorMessage) && (
+        <HeadlessText elementType="div" slot="errorMessage">
```
Comment (KelvinOm) [inline app/client/packages/design-system/widgets/src/components/TagGroup/src/TagGroup.tsx:51]: Each HeadlessText component will create an additional DOM node. Is it necessary? Can we use just WDS Text component in the same way? If not, then may be we can throw WDS typography classes into the spectrum component.

File: app/client/packages/design-system/widgets/src/components/TagGroup/stories/TagGroup.stories.mdx [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+        <Tag id="value-1">Value 1</Tag>
+        <Tag id="value-2">Value 2</Tag>
+        <Tag id="value-3">Value 3</Tag>
+        <Tag id="value-4">Value 4</Tag>
+      </>
+    ),
+  }}
+/>
+
+export const Template = (args) => <TagGroup {...args} />;
```
Comment (KelvinOm) [inline app/client/packages/design-system/widgets/src/components/TagGroup/stories/TagGroup.stories.mdx:19]: There is need to add `ArgsTable`.

File: app/client/packages/design-system/widgets/src/components/TagGroup/stories/TagGroup.stories.mdx [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+  >
+    {Template.bind({})}
+  </Story>
+</Canvas>
+
+# Multiple Selection Mode
+
+Tag Group can be configured to allow multiple selections at a time with the `selectionMode` prop with value `multiple`.
+
+<Canvas>
```
Comment (KelvinOm) [inline app/client/packages/design-system/widgets/src/components/TagGroup/stories/TagGroup.stories.mdx:48]: I can't find how to make that there is no remove button, I also did not notice that the [limit rows](https://react-spectrum.adobe.com/react-spectrum/TagGroup.html#limit-rows) is working.

File: app/client/packages/design-system/widgets/src/components/TagGroup/src/Tag.tsx [hunk 6]
```diff
@@ -0,0 +1,31 @@
+import React from "react";
+import {
+  Tag as HeadlessTag,
+  Button as HeadlessButton,
+} from "react-aria-components";
+import type { TagProps as HeadlessTagProps } from "react-aria-components";
+
+import styles from "./styles.module.css";
+import { CloseIcon } from "../../Modal/src/CloseIcon";
```
[the remark this answers is not in the document]
Comment (KelvinOm) [reply app/client/packages/design-system/widgets/src/components/TagGroup/src/Tag.tsx:9]: He's right about what he says

File: app/client/packages/design-system/widgets/src/components/TagGroup/stories/TagGroup.stories.mdx (hunk 5 above)
Comment (jsartisan) [reply app/client/packages/design-system/widgets/src/components/TagGroup/stories/TagGroup.stories.mdx:56]: Limit rows is not there in hooks or react-aria-components 😞
It needs custom implementation as it depends on the sizing of tags and the container.

File: app/client/packages/design-system/widgets/src/components/TagGroup/src/TagGroup.tsx (hunk 3 above)
Comment (jsartisan) [reply app/client/packages/design-system/widgets/src/components/TagGroup/src/TagGroup.tsx]: Done.

File: app/client/packages/design-system/widgets/src/components/TagGroup/src/Tag.tsx (hunk 2 above)
Comment (jsartisan) [reply app/client/packages/design-system/widgets/src/components/TagGroup/src/Tag.tsx]: Done.

File: app/client/packages/design-system/widgets/src/components/TagGroup/src/Tag.tsx (hunk 1 above)
Comment (jsartisan) [reply app/client/packages/design-system/widgets/src/components/TagGroup/src/Tag.tsx:8]: This is weird for me. It worked without adding it to package.json. I have added it again now.

**Review by KelvinOm — APPROVED** [review]

File: app/client/package.json [hunk 7]
```diff
@@ -385,7 +385,6 @@
     "@blueprintjs/core@^3.47.0": "patch:@blueprintjs/core@npm%3A3.47.0#./.yarn/patches/@blueprintjs-core-npm-3.47.0-a5bc1ea927.patch",
     "@blueprintjs/icons": "3.22.0",
     "@types/react": "^17.0.2",
-    "postcss": "8.4.31",
-    "@react-types/shared": "3.19.0"
```
Comment (KelvinOm) [inline app/client/package.json:389]: ❤️

File: app/client/src/pages/workspace/CreateWorkspaceForm.tsx [hunk 8]
```diff
@@ -42,6 +42,7 @@ export function CreateApplicationForm(
           data-testid="create-workspace-form__name"
           name="name"
           placeholder="Name"
+          // @ts-expect-error Type mismatch
```
Comment (KelvinOm) [inline app/client/src/pages/workspace/CreateWorkspaceForm.tsx:45]: How does this relate to us?

**Review by KelvinOm — CHANGES_REQUESTED** [review]

**Pull request merged**